### ISSUE TITLE
Abstract class / single table discriminator: delete not cascading many-to-one 

### ISSUE THREAD

**Issue by silverbackdan** [doctrine/dbal#3011]
I have an Abstract class with this mapping:
```xml
<doctrine-mapping xmlns="http://doctrine-project.org/schemas/orm/doctrine-mapping"
                  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                  xsi:schemaLocation="http://doctrine-project.org/schemas/orm/doctrine-mapping
                    http://raw.github.com/doctrine/doctrine2/master/doctrine-mapping.xsd"
>
    <entity name="Silverback\ApiComponentBundle\Entity\Content\AbstractContent"
            table="content"
            repository-class="Silverback\ApiComponentBundle\Repository\ContentRepository"
            inheritance-type="SINGLE_TABLE"
    >
        <id name="id" type="string" />
        <discriminator-column name="discr" type="string" />
        <discriminator-map>
            <discriminator-mapping value="page"
                                   class="Silverback\ApiComponentBundle\Entity\Content\Page"
            />
            <discriminator-mapping value="component_group"
                                   class="Silverback\ApiComponentBundle\Entity\Content\ComponentGroup"
            />
        </discriminator-map>
    </entity>
</doctrine-mapping>
```

Then the class extending this is
```xml
<doctrine-mapping xmlns="http://doctrine-project.org/schemas/orm/doctrine-mapping"
                  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                  xsi:schemaLocation="http://doctrine-project.org/schemas/orm/doctrine-mapping
                    http://raw.github.com/doctrine/doctrine2/master/doctrine-mapping.xsd"
>
    <entity name="Silverback\ApiComponentBundle\Entity\Content\Page">
        <many-to-one field="layout" target-entity="Silverback\ApiComponentBundle\Entity\Layout\Layout">
            <cascade><cascade-remove /></cascade>
            <join-column on-delete="CASCADE" />
        </many-to-one>
    </entity>
</doctrine-mapping>
```

When I delete a Layout entity, the Page remains. Is this mapping the correct way around and shouldn't I expect the Page to be deleted when the Layout is?

This is the behaviour while running in a functional test using Behat and a Sqlite database. Is this behaviour supported in Sqlite?

I think the is the right doctrine repo to raise this in but please correct me if not. I'm using the symfony doctrine-bundle which I think means this is the relevant repo.

**Comment by silverbackdan** [comment]
I've been finding more about this. Will continue to investigate looking into tickets such as https://github.com/doctrine/dbal/issues/2833

**Comment by silverbackdan** [comment]
In my case, should the `<cascade><cascade-remove /></cascade>` configuration mean that doctrine should remove the related entities anyway?

I'm looking to also (ideally) detect all the entities that will be removed in a preFlush event. This is so I can check if the entity implements an interface that relates to a file path and to remove the file at the same time.

**Comment by Ocramius** [comment]
```xml
        <many-to-one field="layout" target-entity="Silverback\ApiComponentBundle\Entity\Layout\Layout">
            <cascade><cascade-remove /></cascade>
            <join-column on-delete="CASCADE" />
        </many-to-one>
```

The `cascade-remove` will cause the `Silverback\ApiComponentBundle\Entity\Layout\Layout` to be removed together with the `Silverback\ApiComponentBundle\Entity\Content\Page` that references it when `EntityManager#remove()` and `EntityManager#flush()` are used. 

`<join-column on-delete="CASCADE" />` will define an `ON DELETE="CASCADE"` DDL clause for RDBMSs that support foreign key operations (not yet SQLite - see https://github.com/doctrine/dbal/pull/2836).

Basically, it should work as long as the removal is performed through the ORM.

**Comment by Ocramius** [comment]
@silverbackdan please do send us a test case if what described above doesn't work. See https://github.com/doctrine/doctrine2/tree/master/tests/Doctrine/Tests/ORM/Functional/Ticket for examples.

**Comment by silverbackdan** [comment]
Ah ok, thanks very much @Ocramius - understood!
